### DISCUSSION TITLE
Impossible to use SnowSQL 'authenticator = externalbrowser'

### DISCUSSION THREAD

**Discussion by Igor-Kuzenkov** [theory/sqitch#781] [Q&A]
Hi @sqitch,

- Local dev environment

**OS**: Windows 10
**Sqitch version**: sqitch.bat (App::Sqitch) v1.4.0
**Engine**: Snowflake
**Client**: snowsql.exe

- Intro

Recently in the organization some Snowflake connection restrictions were implemented - IPs were whitelisted.
After this, it's impossible to use _Snowflake_ `username` and `password` from my dev local host in the SnowSQL (client) config.
I need to use my organization email as a `username` and SSO approach to connect to Snowflake.

It works fine when I use SnowSQL as is. Its config looks like (extraction):

_$ ../.snowsql/config_

```
username = first.last@org.com
authenticator = externalbrowser
dbname = ...
rolename = ...
warehouse =...
schema = ...
```
When I run some SnowSQL command (e.g. `snowsql --filename c:\some.sql`), it opens my local browser where I do authentication then it works fine.

- Problem

However, it does not work in conjunction with Sqitch. Sqitch throws the following exception:

_$ sqitch status_

`[Snowflake][DSI] (20032) Required setting 'PWD' is not present in the connection settings. (SQL-28000)`

Does Sqitch support client's `authenticator = externalbrowser` connection approach?
Any suggestions how to solve the problem.

**Comment by theory** [comment]
I don't know. Sqitch itself connects to the database using ODBC but runs the scripts through the SnowSQL client. From the [odbc docs](https://docs.snowflake.com/developer-guide/odbc/odbc-parameters#additional-connection-parameters), it looks like you can specify `authenticator=externalbrowser`; have you tried passing that in the Sqitch URI?

**Comment by Igor-Kuzenkov** [comment]
Just tried but still not working.
This is my `sqitch.conf`:
```
[core]
    engine = snowflake
[engine "snowflake"]
    target = qa
    client = snowsql.exe
[target "qa"]
    uri = "db:snowflake://<snowflake_account>/<SNOWFLAKE_DATABASE_NAME>?Driver=SnowflakeDSIIDriver;warehouse=<SNOWFLAKE_WAREHOUSE_NAME>;role=<SNOWFLAKE_ROLE_NAME>;schema=<SCHEMA>;authenticator=externalbrowser;"
```
Nothing in SnowSQL `config` file, even `[connections]` section is removed.
However, I am getting strange when run `sqitch status` command. It is asking me to log in twice - my browser opens twice. Ok, both times I am entering same creds. But eventually I get exception. Command line and output is:
```
PS C:\...my_sqitch_project_folder...> sqitch status
# On database qa
Initiating login request with your identity provider. A browser window should have opened for you to complete the login. If you can't see it, check existing browser windows, or your OS settings. Press CTRL+C to abort and try again...
Initiating login request with your identity provider. A browser window should have opened for you to complete the login. If you can't see it, check existing browser windows, or your OS settings. Press CTRL+C to abort and try again...
The user you were trying to authenticate as differs from the user currently logged in at the IDP. (SQL-28000)
```
Any ideas?
